compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/FakeOverrideGenerator.kt

```diff
@@ -67,9 +67,21 @@ class FakeOverrideGenerator(
     }
 
     fun IrClass.getFakeOverrides(klass: FirClass<*>, realDeclarations: Collection<FirDeclaration>): List<IrDeclaration> {
+        val classLookupTag = klass.symbol.toLookupTag()
+        val realDeclarationNames = realDeclarations.mapNotNullTo(mutableSetOf()) {
```

Comment:
I have a bad feeling here... What if we have a situation like this
```
open class Base {
    fun foo(s: String) {}
}
class Derived : Base() {
    fun foo() {
        foo("")
    }
}
```

So `foo` without parameter is a real declaration, so name `foo` is filtered and we don't build fake override for `foo` with parameter... What do you think about this case?

Reply:
`allowsToHaveFakeOverrideIn` will return `true` for that case, so it's not filtered out here.

So, the condition below (which returns `null`) is quite strong/safe enough I think, and I left a comment about that: https://github.com/JetBrains/kotlin/pull/4124#discussion_r586132688

Reply:
Just curious, in which cases then will something be filtered out?

Reply:
Those two conditions will return `null` (filter out) if the declaration's dispatch receiver is equal to the containing class (i.e., member) or its visibility is (fake) private. These sound like it will barely not drop anything. :\ I can dig further what were filtered in those 4 modules I saw in full pipeline test, but all the sudden, I'm not confident and not sure it's worth my time. I'm dropping this commit about fake overrides.